include/pybind11/numpy.h

```diff
@@ -646,6 +654,26 @@ class array : public buffer {
         return reinterpret_steal<array>(api.PyArray_Squeeze_(m_ptr));
     }
 
+    /// Resize array to given shape
+    /// If refcheck is true, then resize will fail if > 1 references exists to this array
+    void resize(const std::vector<size_t> &new_shape, bool refcheck = false) {
```

Comment:
Also note that in the [Python API](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.resize.html), refcheck defaults to True.